simapp/app.go

```diff
@@ -197,7 +197,8 @@ func init() {
 // NewSimApp returns a reference to an initialized SimApp.
 func NewSimApp(
 	logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest bool, skipUpgradeHeights map[int64]bool,
-	homePath string, invCheckPeriod uint, encodingConfig simappparams.EncodingConfig, baseAppOptions ...func(*baseapp.BaseApp),
+	homePath string, invCheckPeriod uint, encodingConfig simappparams.EncodingConfig,
+	appOpts servertypes.AppOptions, baseAppOptions ...func(*baseapp.BaseApp),
```

Comment:
Are there any explicit argument we can remove from this constructor that are found in `appOptons`?

Reply:
Not sure. I can handle it in a separate PR.

Reply:
Yes, we can omit `homePath string, invCheckPeriod uint,`

Reply:
I would prefer in another and have this one merged. This one is about `InitChain` options.

Reply:
I added TODOs to https://github.com/cosmos/cosmos-sdk/issues/7178